pkg/admission/types.go

```diff
@@ -31,3 +32,9 @@ type WantsInformerFactory interface {
 	SetInformerFactory(informers.SharedInformerFactory)
 	Validator
 }
+
+// WantsAuthorizer defines a function which sets Authorizer for admission plugins that need it.
+type WantsAuthorizer interface {
+	SetAuthorizer(authorizer.Authorizer)
+	Validator
```

Comment:
IMO I've found the validator interface handy to force folks to remember to implement it even if they don't want to ever return an error.  I see value in making all admission plugins provide `Validate` and moving it out of these "wants" interfaces but then we're still in the same boat of having interfaces that are not as narrow as they possibly could be.